hedera-node/src/main/java/com/hedera/services/sigs/Rationalization.java

```diff
@@ -93,9 +97,7 @@ public SignatureStatus execute() {
             if (allVaryingMaterialEquals(candidateSigs, realSigs) && allStatusesAreKnown(candidateSigs)) {
                 return candidateSigs;
             }
-        } catch (IndexOutOfBoundsException ignore) {
-            log.warn(ignore.getMessage());
-        }
+        } catch (IndexOutOfBoundsException ignore) { }
```

Comment:
Just a question for my understanding: why should we not log if there is error?

Reply:
If there is an `IndexOutOfBoundsException` there, it just means the signature list expanded in the Platform `Transaction` is out-of-date with the new keys, so we have to do synchronous signature verification.

It would be better for performance to not throw and exception and directly check the bounds before calling `txnSigs.subList`...lots of places to improve performance in signature expansion/verification. 😬

Reply:
Ah Understood. Thanks @tinker-michaelj  !!  Yeah totally agreed :)